pages/board.tsx

```diff
@@ -152,23 +159,90 @@ export default class BoardPage extends React.Component<Props, State> {
     );
   }
 
+  _renderBoard(data: Board) {
+    return (
+      <MoveCardMutation mutation={MOVE_CARD}>
+        {_moveCard => {
+          return (
+            <div className="mom-board">
+              <DragDropContext 
+                onDragEnd={(e) => {
+                  _moveCard({
+                      variables: {
+                        id: e.draggableId,
+                        columnId: e.destination.droppableId
+                      },
+                      optimisticResponse: {
+                        updateCard: {
+                         __typename: "UpdateCardPayload",
+                         card: {
+                          __typename: "Card",
+                          id: e.draggableId,
+                          column: {
+                            __typename: "Column",
+                            id: e.destination.droppableId
+                          }
+                         }
+                        }
+                      },
+                      update: (proxy, { data: { updateCard } }) => {
+                        // Read the data from our cache for this query.
+                        const data = proxy.readQuery({
+                          query: GET_BOARD,
+                          variables: { id: this.props.id },
+                        });
+
+                        let card = null;
+                       
+                        for(let i=0; i<data['board'].columns.length; i++) {
+                          card =  data['board'].columns[i].cards.filter(card => card.id === updateCard.card.id)
+                         
+                          data['board'].columns[i].cards.splice(card.id, 1);
```

Comment:
The code here is using the id of the card as its index in the call to `splice` which wont be working.